src/encoded/static/components/rnaget.js

```diff
@@ -0,0 +1,889 @@
+        return value.map((relatedSeries) => relatedSeries['@id']).join();
+    }
+    return null;
+};
+
+RNAGetDataRegistry.register({ field: 'related_series', type: 'Experiment' }, ExperimentRelatedSeriesRenderer);
+
+
+/**
+ * `thumb_nail` renderer for type=Image. Displays an image thumbnail that links to a full-size
+ * image.
+ */
+const ImageThumbnailRenderer = ({ value }) => (
+    <div className="tcell-thumbnail">
+        <a href={value} target="_blank" rel="noopener noreferrer">
+            <img src={value} alt={value} />
+        </a>
+    </div>
+);
+
+ImageThumbnailRenderer.propTypes = {
+    /** Path of image to display */
+    value: PropTypes.string.isRequired,
+};
+
+RNAGetDataRegistry.register({ field: 'thumb_nail', type: 'Image' }, ImageThumbnailRenderer);
+
+
+/**
+ * Displays the image download_url path as a link to the image.
+ */
+const ImageDownloadRenderer = ({ value }, reactContext) => {
+    const parsedUrl = url.parse(reactContext.location_href);
+    return <a href={`${parsedUrl.protocol}//${parsedUrl.host}${value}`}>{value}</a>;
+};
+
+ImageDownloadRenderer.propTypes = {
+    /** path to image */
+    value: PropTypes.string.isRequired,
+};
+
+ImageDownloadRenderer.contextTypes = {
+    location_href: PropTypes.string,
+};
+
+RNAGetDataRegistry.register({ field: 'download_url', type: 'Image' }, ImageDownloadRenderer);
+
+
+/**
+ * `files` renderer for type=Experiment. Displays a comma separated list of accessions that link to
+ * the corresponding file page.
+ */
+const ExperimentFilesRenderer = ({ item }) => {
+    if (item.files && item.files.length > 0) {
+        return item.files.map((file) => <a key={file['@id']} href={file['@id']}>{globals.atIdToAccession(file['@id'])}</a>).reduce((prev, curr) => [prev, ', ', curr]);
+    }
+    return null;
+};
+
+ExperimentFilesRenderer.propTypes = {
+    /** Object (row) currently rendered */
+    item: PropTypes.object.isRequired,
+};
+
+RNAGetDataRegistry.register({ field: 'files.@id', type: 'Experiment' }, ExperimentFilesRenderer);
+
+
+/**
+ * Make complete URLs from the file download paths.
+ */
+const FileDownloadHref = ({ value }, reactContext) => {
+    const parsedUrl = url.parse(reactContext.location_href);
+    return `${parsedUrl.protocol}//${parsedUrl.host}${value}`;
+};
+
+FileDownloadHref.contextTypes = {
+    location_href: PropTypes.string,
+};
+
+RNAGetDataRegistry.register({ field: 'href', type: 'File' }, FileDownloadHref);
+
+
+/**
+ * Get the property specified by a field from the search results and return a string from that.
+ */
+const lookupColumn = (result, column) => {
+    let nodes = [result];
+    const names = column.split('.');
+
+    for (let i = 0, len = names.length; i < len && nodes.length > 0; i += 1) {
+        let nextnodes = [];
+        _.each(nodes.map((node) => node[names[i]]), (v) => {
+            if (v === undefined) return;
+            if (Array.isArray(v)) {
```

Comment:
Code smell. In Javascript, you search for functionality like length. You treat object and arrays as collections and like they act the same way. When someone looks specifically for types, my spider senses start tingling.